generator/internal/sources_generator.h

```diff
@@ -0,0 +1,74 @@
+// Copyright 2024 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef GOOGLE_CLOUD_CPP_GENERATOR_INTERNAL_SOURCES_GENERATOR_H
+#define GOOGLE_CLOUD_CPP_GENERATOR_INTERNAL_SOURCES_GENERATOR_H
+
+#include "generator/internal/service_code_generator.h"
+#include "google/cloud/status.h"
+#include <google/protobuf/compiler/code_generator.h>
+#include <google/protobuf/descriptor.h>
+#include <map>
+#include <string>
+
+namespace google {
+namespace cloud {
+namespace generator_internal {
+
+/**
+ * Generates the conglomerate source file for a service.
+ *
+ * This conglomerate source is essentially a concatenation of the other source
+ * files emitted by the generator, for a service.
+ *
+ * Every translation unit (TU) that transitively includes a protobuf header
+ * needs to recompile that header. To minimize build times, it is best to have
+ * just one TU.
+ *
+ * See go/cloud-cxx:reducing-build-times for further discussion on the design,
+ * plus some experimental results.
```

Comment:
Don't name internal documents?

Reply:
We do this in some other places in the code.

Reply:
OSS code and internal documents don't exactly mix. I would always avoid linking internal documents in public documentation.

This documentation is internal though. It is for us. I like having a link to why we thought this was a good idea, even if some of us may not be able to access it tomorrow.